src/sentry/integrations/opsgenie/utils.py

```diff
@@ -1,13 +1,87 @@
+import logging
 from typing import Optional
 
+from sentry.constants import ObjectStatus
+from sentry.incidents.models import AlertRuleTriggerAction, Incident, IncidentStatus
+from sentry.integrations.metric_alerts import incident_attachment_info
+from sentry.services.hybrid_cloud.integration import integration_service
 from sentry.services.hybrid_cloud.integration.model import RpcOrganizationIntegration
+from sentry.shared_integrations.exceptions import ApiError
+
+logger = logging.getLogger("sentry.integrations.opsgenie")
+from .client import OpsgenieClient
+
+
+def build_incident_attachment(incident, new_status: IncidentStatus, metric_value=None):
+    data = incident_attachment_info(incident, new_status, metric_value)
+    alert_key = f"incident_{incident.organization_id}_{incident.identifier}"
+    if new_status == IncidentStatus.CLOSED:
+        payload = {"identifier": alert_key}
+    priority = "P1"
+    if new_status == IncidentStatus.WARNING:
+        priority = "P2"
```

Comment:
How does this 'P2' priority get used? It appears to be set, but the only occurance it's used in is then skipped over by this conditional.

Maybe these lines should be after `alert_key = `, and then the payload declaration should be after, without an `else` block?

Reply:
Oh wait, I'm silly